design_docs/generalize_calendar.md

```diff
@@ -0,0 +1,249 @@
+<h1> Title: Generalize Calendar supported by Analysis <br>
+Xylar Asay-Davis <br>
+date: 2017/02/09 <br>
+</h1>
+<h2> Summary </h2>
+Currently, the time variable in `xarray` data sets within MPAS-Analysis have two
+major shortcomings, inherited from `xarray` (through `pandas` and `numpy.datetime64`).
+First, only the Gregorian calendar is supported.  Second, there is not support
+for dates outside the years 1678 to 2262.  The analysis needs to support both
+the Gregorain  ('gregorian') and the 365-day ('gregorian_noleap') calendars.  It also needs to
+support, at a minimum, years between 0001 and 9999, and preferably arbitrary
+years both positive and negative.
+
+A major challenge is that it seems that xarray cannot easily be forced to
+use an alternative representation of dates to the troublesome
+`numpy.datetime64` type.  The most obvious alternative, `datetime.datetime`,
+seemingly cannot be useddirectly in `xarray` because objects of this type
+are converted to `numpy.datetime64` objects at various stages when using 
+features from pandas, raising errors when dates are out of range.  While an
+alternative date class (e.g. `netcdftime.DatetimNoLeap`) might be used to
+represent dates on the 'gregorian_noleap' calendar, there is no such
+preexisting alternative for the 'gregorian' calendar.
+
+The solution proposed herein is to store time as floating-point days since the
+reference date 0001-01-01 and to convert dates in this format to 
+`datetime.datetime` and `MpasRelativeDelta` objects whenever mathematical
+manipulation of dates is required.
+
+A successful implementation would produce essentially identical analysis to
+what is currently produced, but making use of the dates from the MPAS calendar
+(whether Gregorian or 365-day) without the need for artifical offsets (e.g.
+`yearOffset` used in the current code.  Plots of horizontal fields would remain
+unchanged while plots of time series would have a time axis with the simulation
+date instead of the offset date.
+
+
+<h1> Requirements </h1>
+
+<h2> Requirement: The 'Time' coordinate of xarray data sets must be consistent 
+with the MPAS calendar <br>
+Date last modified: 2017/02/09 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+For all data sets used in the analysis, the 'Time' coordinate must represent dates 
+on the appropriate MPAS calendar, either 'gregorain' or 'gregorain_noleap', depending
+on the namelist option 'config_calendar_type'.  There must be ways of mathematically
+manipulating times (e.g. adding/subtracting offsets and figuring out the amount of time
+between two dates) and of making plots that are consistent with these calendars.
```

Comment:
Regarding the second issue, there's a function `plt.plot_date` that takes exactly the format I'm suggesting here: days since 0001-01-01.  It doesn't support other reference dates, so that was one reason I chose that reference.